### DISCUSSION TITLE
Can this be a SQL Delight Replacement

### DISCUSSION THREAD

**Discussion by criticalstrike18** [tursodatabase/libsql#1815] [Ideas]
This is specifically for KMM as sqldelight  doesn't support the batch Transaction and room has not support for desktop or ios.

**Comment by marcardar** [comment]
Hmm, I was thinking SQLDelight would be used on top of libsql, and that libsql is a replacement for using androidx SupportSQLiteDatabase. Regardless of batch transaction support, did you try to get them working together?

**Reply by criticalstrike18** [reply]
yeh libsql is overall doing all the work while sqldelight is the tool for kmm it works for me right now just needs different drivers according to the overall system But libsql is not a replacement of androidx SupportSQLiteDatabase. as supportsql offers various types of impl libsql is maybe one of them is my assumption idk the actual scenario

**Reply by marcardar** [reply]
So, my app is already using SQLDelight with androidx SupportSQLiteDatabase. Apparently, a new driver needs to be implemented for SQLDelight to be able to use libsql. I haven't yet looked into how feasible this is, and was hoping someone had already done it!

**Reply by criticalstrike18** [reply]
can you refrence your code if its available freely i would like to look into it to understand the scenario

**Reply by marcardar** [reply]
It's not freely available, but anyway there's nothing special about it. It's standard use of SQLDelight in an Android kotlin app.
